rules/Symfony64/Rector/Class_/FixImportWhileUsingRouteAttributeButAnnotationImportRector.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Symfony\Symfony64\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Attribute;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Symfony\Tests\Symfony64\Rector\Class_\FixImportWhileUsingRouteAttributeButAnnotationImportRector
+ */
+final class FixImportWhileUsingRouteAttributeButAnnotationImportRector extends AbstractRector
+{
+    private const ANNOTATION_ROUTE = 'Symfony\Component\Routing\Annotation\Route';
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace Symfony\Component\Routing\Annotation\Route by Symfony\Component\Routing\Attribute\Route when the class use #Route[] attribute',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+    use Symfony\Component\Routing\Annotation\Route;
+
+    /**
+     * #[Route("/foo")]
+    */
+    public function create(Request $request): Response
+    {
+        return new Response();
+    }
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+    use Symfony\Component\Routing\Attribute\Route;
+
+    #[Route('/foo')]
+    public function create(Request $request): Response
+    {
+        return new Response();
+    }
+CODE_SAMPLE
+                ),
+
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [Attribute::class];
+    }
+
+    /**
+     * @param Attribute $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node instanceof Attribute && $node->name !== null && (string) $node->name === self::ANNOTATION_ROUTE) {
+            // there I'd like to modify the UseUse node representing my current node->name
+            // AFAIK the node I'm returning from this method is only the Attribute node
+            // maybe should I retrieve the global AST from somewhere and manually get the UseUse node to edit the value
+            // Or maybe there is a more elegant way to do that ?
+
+            return null;
```

Comment:
I would like some help on this part. I think this if statement is enough to identify that the #[Route] attribute is used in the file.

But I'm stuck to modify the related UseUse node of this attribute. Have you any recommendations on this ?

Reply:
No need to import to use statement, that will be taken care by : `$rectorConfig->importNames()` when enabled.

Reply:
Oh I see what you mean, in that case, `$rectorConfig->importNames()` is working on `RenameClassRector`, but not this custom rector, you may need to lookup of what AnnotationToAttributeRector implementation, set Name node, and add Name node to import nodes collector to be imported later

Reply:
You can use `Rector\PostRector\Collector\UseNodesToAddCollector` service for that, eg:

```php
$this->useNodesToAddCollector->addUseImport(new FullyQualifiedObjectType($namespacedAttrName));
```

see https://github.com/rectorphp/rector-src/blob/663f526966313e0f178c088ea667119490404da9/rules/Php80/Rector/Property/NestedAnnotationToAttributeRector.php#L219

Reply:
After some thinking, replace use statement directly is unnecessary on this case, if that needed, you can do:

```php
$rectorConfig->importNames();
$rectorConfig->removeUnusedImports();
```

That will be safer imo, and that will be replaced.

The improvement for this case need to be verified carefully, and that need separate PR.